system/jlib/jfile.cpp

```diff
@@ -7106,3 +7109,230 @@ const FileSystemProperties & queryFileSystemProperties(const char * filename)
     else
         return linuxFileSystemProperties;
 }
+
+
+#ifndef _WIN32
+class CFileEventWatcher : public CInterfaceOf<IFileEventWatcher>, implements IThreaded
+{
+    class CMonitorItem : public CInterface
+    {
+    public:
+        CMonitorItem(const char *_filename, int _watchFd, FileWatchEvents _events) : filename(_filename), watchFd(_watchFd), events(_events)
+        {
+        }
+        std::string filename;
```

Comment:
StringAttr might be better - I assume immutable

Reply:
std::string easier to use with other stl containers.